src/share/classes/com/sun/javatest/regtest/util/FileUtils.java

```diff
@@ -0,0 +1,155 @@
+import java.util.stream.Stream;
+
+/**
+ * Utilities for handling files.
+ */
+public class FileUtils {
+    /**
+     * An unchecked IO exception to be thrown when various NIO file operations fail,
+     * and throw a checked IO exception.
+     */
+    public static class NIOFileOperationException extends UncheckedIOException {
+        private static final long serialVersionUID = 0;
+
+        /**
+         * The operation that caused an exception.
+         */
+        enum Op {MKDIRS, LAST_MOD, LIST, SIZE}
+
+        /**
+         * Creates an unchecked IO exception for a checked IO exception
+         * that was caused by an operation on a file.
+         *
+         * @param op    the operation
+         * @param p     the file
+         * @param cause the IO exception
+         */
+        NIOFileOperationException(Op op, Path p, IOException cause) {
+            super(getMessage(op, p), cause);
+        }
+
+        private static String getMessage(Op op, Path p) {
+            switch (op) {
+                case MKDIRS:
+                    return "Cannot create directories for " + p;
+                case LAST_MOD:
+                    return "Cannot access last-modified time for " + p;
+                case LIST:
+                    return "Cannot list directory " + p;
+                case SIZE:
+                    return "Cannot determine size of file " + p;
+                default:
+                    return "Cannot perform unknown operation for " + p;
+            }
+        }
+    }
+
+    private FileUtils() { }
+
+    /**
+     * Returns the size of a file, in  bytes.
+     *
+     * @param p the file
+     * @return the size of a file, in  bytes
+     * @throws NIOFileOperationException if an IO exception arose while accessing the size
+     */
+    public static long size(Path p) throws NIOFileOperationException {
+        try {
+            return Files.size(p);
+        } catch (IOException e) {
+            throw new NIOFileOperationException(NIOFileOperationException.Op.SIZE, p, e);
+        }
+    }
+
+    /**
+     * Returns the time that a file was last modified.
+     *
+     * @param p the file
+     * @return the time the file was last modified
+     * @throws NIOFileOperationException if an IO exception arose while accessing the time
+     */
+    public static FileTime getLastModifiedTime(Path p) throws NIOFileOperationException {
+        try {
+            return Files.getLastModifiedTime(p);
+        } catch (IOException e) {
+            throw new NIOFileOperationException(NIOFileOperationException.Op.LAST_MOD, p, e);
+        }
+    }
+
+    /**
+     * Compares the times that a pair of files were last modified.
+     *
+     * @param p1 the first file
+     * @param p2 the second file
+     * @return {@code -1}, {@code 0} or {@code 1} according to whether the last modified time of the first file
+     *          is less than, equal to, or greater than the last modified time of the second file
+     *
```

Comment:
Not sure I understand the spacing convention, but the presence of this empty line and those at lines 109, 111, 113, and 128 may be inconsistent.

Reply:
Inconsistency is bad, especially within a new file. I was aiming for a more spaced-out look, more similar to the core-libs style, with blank lines between groups of tags.